### PR TITLE
Allow for use of privateKey instead of privateCert

### PR DESCRIPTION
**Pull request by alon85**
An updated PR to address the conversation here: https://github.com/node-saml/passport-saml/pull/387

### CODE REVIEW

**Review by markstos — APPROVED** [review]
This has code/docs/tests and is backward compatible. It looks good to me. I updated the docs to further simplify them and focus on the clearer `privateKey` name instead of the less clear "private cert". 

Assigning to @cjbarth for a second look.

File: src/passport-saml/saml-post-signing.ts [hunk 1]
```diff
@@ -15,15 +16,15 @@ interface SignSamlPostOptions {
 export function signSamlPost(samlMessage: string, xpath: string, options: SignSamlPostOptions) {
   if (!samlMessage) throw new Error('samlMessage is required');
   if (!xpath) throw new Error('xpath is required');
-  if (!options || !options.privateCert) throw new Error('options.privateCert is required');
+  if (!options || (!options.privateCert && !options.privateKey)) throw new Error('options.privateCert or options.privateKey is required');
```
Comment (cjbarth) [inline src/passport-saml/saml-post-signing.ts:19]: What do you think of issuing a deprecation warning if someone is using `privateCert`? This way, we can remove this extra code with a future semver-major release.

**Review by cjbarth — APPROVED** [review]
I just have one suggestion, but I don't think it should block this merge.

**Review by RyanCopley — APPROVED** [review]
:wave: Thank you Alon! I hope you are doing well :)

File: src/passport-saml/saml-post-signing.ts (hunk 1 above)
Comment (markstos) [reply src/passport-saml/saml-post-signing.ts:19]: @cjbarth That's OK if you want to add the warning.

**Comment by markstos** [comment]
I'm going to go ahead and merge this. We can add a deprecation warning in a future release. It will clean up the code to eventually not support both names.

**Pull request merged**